### DISCUSSION TITLE
Mount lifecycle is called twice and listener 4 times

### DISCUSSION THREAD

**Discussion by OlrikNijmeijer** [livewire/livewire#6863] [7. Bugs (archived)]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I upgraded a project from livewire v2 to livewire v3.
After this i noticed that the mount function of a component is called twice.

Also when I create a link that dispatched an event, the listener is called 4 times.
This also happens when i create a new page with an empty layout.

**Component**
```
namespace App\Livewire\Pages\Portal;

use Livewire\Component;

class TestPage extends Component
{
	protected $listeners = ['doSomething'];

	public function mount()
	{
		dump('mount called');
	}

	public function render()
	{
		return view('livewire.pages.portal.test-page');
	}

	public function dispatchSomething()
	{
		$this->dispatch('doSomething');
	}

	public function doSomething(): void
	{
		dump('doSomething called');
	}
}
```

**Blade**
```
<div>
    <x-button wire:click="dispatchSomething" label="test button"/>
</div>
```

**Layout**
```
<!doctype html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}" class="light">
<body>
{{$slot}}
</body>
</html>
```

This results in the following views:

**Dump in mount**

![Scherm­afbeelding 2023-09-19 om 14 48 02](https://github.com/livewire/livewire/assets/17452023/0591499c-1a21-444e-83cb-2f97bf4470df)

**Dump in listener**

![Scherm­afbeelding 2023-09-19 om 14 48 07](https://github.com/livewire/livewire/assets/17452023/f01ad10a-0d29-46d6-b4f2-6e753e9eb543)

I created a new Laravel project to try to reproduce this, but in the new project this problem does not appear.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by jacksontong** [comment]
are you using [https://github.com/rappasoft/laravel-livewire-tables](https://github.com/rappasoft/laravel-livewire-tables)? I have same issue because of https://github.com/rappasoft/laravel-livewire-tables/issues/1353

**Reply by OlrikNijmeijer** [reply]
Yes i do use that table package. If i disable the "inject_assets" option in the config file, this issue does not appear anymore.
But after I do that, the tables also don't work anymore.

I see a reply in the issue you send that there will be a fix soon, so hopefully this solves the issue.

**Comment by renky** [comment]
I had a similar issue in a package of me. It is kind of something like rappasoft livewire-tables... In my case all trait-hooks have been called twice for my own "WithPagination" trait. I have seen that rappasoft did the same as me: Rappasoft uses a trait called WithPagination that again uses the original Livewire WithPagination trait.
so finally the hooks are called one time for the original WithPagination Livewire trait and then for my own trait. After renaming my own Trait to "WithPages" the issue was gone..

Maybe that helps finding the issue - I don't believe that it is really a Livewire-Ission - except if we want to care about such edge cases where traits use other traits with the same Basename - it would be possible to detect it and only mount the latest one, but on the other hand I think it sould be solved by the traits itself.
